cmd/kcp-front-proxy/main.go

```diff
@@ -40,26 +66,39 @@ forwards Common Name and Organizations to backend API servers in HTTP headers.
 The proxy terminates TLS and communicates with API servers via mTLS. Traffic is
 routed based on paths.`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return server.Serve()
+			if err := options.Complete(); err != nil {
+				return err
+			}
+			if errs := options.Validate(); errs != nil {
+				return errors.NewAggregate(errs)
+			}
+			proxyHandler, err := proxy.NewHandler(&options.Proxy)
+			if err != nil {
+				return err
+			}
+
+			var servingInfo *genericapiserver.SecureServingInfo
+			var loopbackClientConfig *restclient.Config
+			if err := options.SecureServing.ApplyTo(&servingInfo, &loopbackClientConfig); err != nil {
+				return err
+			}
+			doneCh, err := servingInfo.Serve(proxyHandler, time.Second*60, ctx.Done())
+			if err != nil {
+				return err
+			}
+
+			<-doneCh
+			return nil
 		},
 	}
 
-	// setup klog
-	fs := goflags.NewFlagSet("klog", goflags.PanicOnError)
-	klog.InitFlags(fs)
-	cmd.PersistentFlags().AddGoFlagSet(fs)
+	options.AddFlags(cmd.Flags())
 
-	cmd.PersistentFlags().StringVar(&server.ListenAddress, "listen-address", ":8083", "Address and port for the proxy to listen on")
-	cmd.PersistentFlags().StringVar(&server.ClientCACert, "client-ca-cert", "certs/kcp-client-ca-cert.pem", "CA cert used to validate client certs")
```

Comment:
@sttts The `--client-ca-file` option isn't available to verify client certs.

Reply:
that's via authentication. Have not wired that, but left your code in place. Might be worth to explore using the kube authenticators too.

Reply:
Would be good to know whether I broke the old code or whether it still works. We can attempt to use kube's client cert auth in follow-up.

Reply:
The old code was validating client certs, so it's broken. :) I chatted with @ncdc and got some pointers for wiring up kube's client cert auth, though. I'll merge and start on a follow up.